src/datachain/query/dataset.py

```diff
@@ -431,6 +432,18 @@ def populate_udf_table(self, udf_table: "Table", query: Select) -> None:
 
         processes = determine_processes(self.parallel)
 
+        count_query = sqlalchemy.select(f.count()).select_from(query.subquery())
+        rows_total = next(self.catalog.warehouse.db.execute(count_query))[0]
+
+        if rows_total == 0:
+            return
+        if rows_total == 1:
+            # Single row optimization to avoid unnecessary processing
+            # this is useful then using `from_storage` function since it always (?)
+            # will be only one row with bucket URI to generate rows from
+            workers = False
+            processes = False
```

Comment:
This optimization will skip distributed compute mode for UDFs with only one row.

Let's say we have `workers` settings in `DATACHAIN_SETTINGS_WORKERS` env variable, and number of workers is set to 4.

Then for `from_storage` function distributed compute will be running, but it will run only one worker inside, instead of 4 workers, with all heavy distributed compute machinery running.